tests/test_diffraction_objects.py

```diff
@@ -254,7 +257,10 @@ def test_init_invalid_xtype():
             },
             {"xtype": "q", "yarray": np.array([1, 2, 4, 6])},
         ),
-        (  # C4: Different x-array lengths with approximate x-value match
+        (  # C4: Different x-array lengths
```

Comment:
okay - clear, I understand the function now. yeah for C2, it was hard for me to understand what the function does by reading the test alone since numbers were similiar but other cases are clear.

Reply:
Thanks! I'll edit C2